autoit_unpack.py

```diff
@@ -90,7 +114,87 @@ def deassemble_script(script_data: bytes) -> str:
     return out
 
 
-def parse_au3_header(data: bytes) -> Optional[Tuple[int, Optional[str]]]:
+def parse_au3_header_ea05(
+    data: bytes, checksum: int
+) -> Optional[Tuple[int, Optional[str]]]:
+    off = 0
+    file_str = decrypt_mt(data[off:][:4], 0x16FA)
+    if file_str != b"FILE":
+        return None
+
+    off += 4
+    flag = struct.unpack("<I", data[off:][:4])[0] ^ 0x29BC
```

Comment:
A bit ugly :thinking:. I understand there is no malduck dependency, but maybe :smol: helper

```python
def u32(data: bytes) -> int:
    return struct.unpack("<I", data)
```

wouldn't hurt?

OTOH it's a reversing tool, not a code beauty contest, so YMMV.

Reply:
(but following the struture is not trivial due to this and incrementing offset i guess)